cmd/charmd/main.go

```diff
@@ -116,9 +116,18 @@ func serve(confPath string) error {
 	if err != nil {
 		return errgo.Notef(err, "cannot create new server at %q", conf.APIAddr)
 	}
-
+	handler := server.(http.Handler)
```

Comment:
I wanted to do that, but NewServer returns an HTTPCloser and so its a different type than CombinedLoggingHandler returns and so I needed to use a different variable. I took a look a net/http to make sure it doesn't call close and I found nothing so I'm pretty sure we only have this HTTPCloser for testing.